src/main/java/slimeknights/mantle/client/book/HTMLUtils.java

```diff
@@ -0,0 +1,89 @@
+package slimeknights.mantle.client.book;
+
+
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.TextColor;
+
+import javax.annotation.Nullable;
+import java.util.stream.Collectors;
+
+public class HTMLUtils {
+
+  public static String line(String text, String... styles) {
+    return line(text, false, styles);
+  }
+
+  public static String line(String text, boolean underline, String... styles) {
+    return line(text, null, underline, false, styles);
+  }
+
+  public static String line(String text, boolean underline, boolean large, String... styles) {
+    return line(text, null, underline, large, styles);
+  }
+
+  public static String line(String text, @Nullable String id, boolean underline, boolean large, String... styles) {
+    StringBuilder builder = new StringBuilder("<p");
+
+    if (id != null) builder.append(" id=\"").append(id).append("\"");
+
+    if (underline || large) builder.append(" class=\"");
+    if (underline) builder.append("underline ");
+    if (large) builder.append("large");
+    if (underline || large) builder.append("\"");
+
+    if (styles.length > 0) builder.append(" style=\"").append(String.join("; ", styles)).append("\"");
+
+    return builder.append(">")
+      .append(text)
+      .append("</p>")
+      .toString();
+  }
+
+  public static String line(Component component) {
+    return line(toHTML(component));
+  }
+
+  /**
+   * NOTE: uses a span instead of p to recursively inline
+   */
+  private static String toHTML(Component component) {
+    StringBuilder builder = new StringBuilder("<span");
+
+    Style style = component.getStyle();
+    if (!style.isEmpty()) {
+      builder.append(" style=\"");
+
+      TextColor color = style.getColor();
+      if (color != null && color.getValue() != 0) {
+        builder.append("color: ");
+        // toString gives a weird color
+        builder.append(hexRGB(color.getValue()));
+        builder.append(";");
+      }
+
+      // TODO: I don't think these are ever used in the in game book
+//      if (style.isBold()) {}
+//      if (style.isItalic()) {}
+//      if (style.isStrikethrough()) {}
+//      if (style.isUnderlined()) {}
```

Comment:
They are used in Metalborn's book, and in a few tinkers elements I'm pretty sure.